lib/EarlyFunctionAnalysis/EarlyFunctionAnalysis.cpp

```diff
@@ -1841,10 +1994,18 @@ bool EarlyFunctionAnalysis::runOnModule(Module &M) {
   for (const auto &[_, Node] : BasicBlockNodeMap)
     RootNode->addSuccessor(Node);
 
-  UniquedQueue<BasicBlockNode *> CFEPQueue;
+  // Create an over-approximated call graph of the program. A queue of all the
+  // function entrypoints is maintained.
+  BasicBlockQueueTy FunctionEntrypointsQueue;
   for (auto *Node : llvm::post_order(&CG)) {
-    if (Node != RootNode)
-      CFEPQueue.insert(Node);
+    if (Node == RootNode)
+      continue;
+
+    // The intraprocedural analysis will be scheduled only for those functions
+    // which have `Invalid` as type.
+    auto &Function = Binary.Functions.at(getBasicBlockPC(Node->BB));
+    if (Function.Type == model::FunctionType::Invalid)
+      FunctionEntrypointsQueue.insert(Node);
   }
```

Comment:
`FunctionEntrypointsQueue` is used only by `Analyzer`. I think you can push it to become a member of its class and initialize it in the constructor or somewhere.